modules/api/docs/LorisRESTAPI_v0.0.4-dev.md

```diff
@@ -823,90 +823,191 @@ The JSON object is of the form:
-                }]
+                }
+            ]
         },
         {
-            "Tarname" : "DCM_yyyy-mm-dd_ImagingUpload-hh-mm-def456.tar",
-            "Patientname" : "DCM456_654321_V1",
-            "SeriesInfo" :
-                [{
-                "SeriesDescription" : "MPRAGE_ipat2",
-                "SeriesNumber" : "2",
-                "EchoTime" : "2.98",
-                "RepetitionTime" : "2300",
-                "InversionTime" : "900",
-                "SliceThickness" : "1",
-                "Modality" : "MR",
-                "SeriesUID" : "1.7.8.9.1296",
-                }]
+            "ArchiveName" : "DCM_yyyy-mm-dd_ImagingUpload-hh-mm-def456.tar",
+            "PatientName" : "DCM456_654321_V1",
+            "SeriesInfo" : [
+                {
+                  "SeriesDescription" : "MPRAGE_ipat2",
+                  "SeriesNumber" : 2,
+                  "EchoTime" : 2.98,
+                  "RepetitionTime" : 2300,
+                  "InversionTime" : 900,
+                  "SliceThickness" : 1,
+                  "Modality" : "MR",
+                  "SeriesUID" : "1.7.8.9.1296",
+                }
+            ]
         }
-    ]    
+    ]
 }
 ```
 
-The `Modality` header in the SeriesInfo is either `MR` or `PT` for MRI or PET 
+The `Modality` header in the SeriesInfo is either `MR` or `PT` for MRI or PET
 scans, respectively.
 
+```
+POST /candidates/$CandID/$Visit/dicoms
+```
+
+In addition to the uploaded file, the body of the POST request should contain a
+`json` form attribute containing the following keys and values:
+
+```js
+{
+    "CandID": $CandID,
+    "PSCID": $PSCID,
+    "Visit": $Visit,
+    "IsPhantom": boolean
+}
+```
+
+An optionnal header 'LORIS-Overwrite' with the value 'overwrite' can be used to
+overwrite an existing file.
+A successful request will be answered by a `303 See Other` response with its
+`Location` header pointing to the processes list of the new upload.
+(See 5.3 Tar Level processes)
+
 ### 5.2 Tar Level Data
+
+This section describe how to upload DICOM studies and how to start and monitor
+`mri_upload` processes.
+DICOM studies that have been successfully uploaded and processed can be downloaded with
+the following `GET` request:
+
 ```
-GET /candidates/$CandID/$VisitLabel/dicoms/$Tarname
+GET /candidates/$CandID/$VisitLabel/dicoms/$ArchiveName
 ```
 
-Returns/Downloads a `tar` file which contains a `.meta` and a `.log` text 
+Returns/Downloads a `tar` file which contains a `.meta` and a `.log` text
 files, and a `.tar.gz` of the raw DICOM data as acquired during the candidate
 scanning session, and as retrieved from `/candidates/$CandID/$Visit/dicoms`.
 
-Only `GET` is currently supported.
+To get a list of the processes and their status for a given DICOM study previously uploaded use the following:
+
+```
+GET /candidates/$CandID/$VisitLabel/dicoms/$Tarname/processes
+```
+
+The response contains all `mri_upload` attempts with the specified `$tarname`. And for
```

Comment:
technically, there should only be one `UploadID` per `TarchiveID`. So when you overwrite, you should grep the `UploadID` associated to what is being overwritten instead of inserting a new entry in `mri_upload`. Especially since the incoming directory content will be overwritten anyway.

Reply:
I agree that there is only one MRI upload per DICOM archive, but I've seen a few cases in testing (even without this PR) where there are a few failed MRI uploads and one successful for a single file. This is how the batch uploader does it (don't know about other pipelines).

Reply:
yeah. I think in principle, there should only be one upload per tarchive but unfortunately the batch upload script is not compliant with this set up. However, for the API, I think we should make sure this is the behaviour at minimum, to avoid duplicated entries.

Reply:
Is this thread resolved?